lib/msf/core/post/file.rb

```diff
@@ -361,6 +361,18 @@ def upload_file(remote, local)
     write_file(remote, ::File.read(local))
   end
 
+  #
+  # Sets the executable permissions on a remote file
+  #
+  # @param remote [String] Destination file name on the remote filesystem
+  def chmod_x_file(remote)
+    if session.type == "meterpreter" and session.commands.include?('stdapi_fs_chmod')
+      session.fs.file.chmod(remote, 0700)
```

Comment:
Is there a reason `700` is used over `+x` ?

Reply:
For the cmd_exec or for meterpreter?
For cmd_exec it's for compatibility (I think busybox doesn't support +x, only 700).
For meterpreter it's because +x is not implemented (yet).

Reply:
BusyBox definitely supports symbolic modes. Meterpreter is the bigger concern. Who wants to write that? :-)

I chose `700` because I usually want just my current user to have access to a file (maybe it's a privesc). That may not be the case everywhere.

Reply:
I would prefer to be able to set `+x` but can't immediately think of any viable scenarios where `0700` is inferior.